### PR TITLE
Integration specs for views

### PR DESCRIPTION
**Pull request by JuanLPalacios**
integration tests for:

- log in
- users index
- users show
- user's posts index
- user's posts show

### CODE REVIEW

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @JuanLPalacios 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️

image

 
 Please fix the tests that are failing.
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by JuanLPalacios** [comment]
@Emmyn5600  that is not an error if the location is /users/sign_in instead of / it means login failed.
I would kindly ask you to check if you seeded before running tests
```
rails db:create
rails db:schema:load
rails db:seed
```

File: spec/features/log_in_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  it 'When I click the submit button after filling in the username and ' \
+     'the password with correct data, I am redirected to the root page.' do
+    fill_in('user_email', with: 'test0@test.com')
+    fill_in('user_password', with: 'password')
+    click_button 'Log in'
+
+    expect(page).to have_current_path(root_path)
+  end
+end
```
Comment (ahmetbozaci) [inline spec/features/log_in_spec.rb]: Screenshot from 2022-07-02 22-52-26

- [ ] Like the last code reviewer said kindly fix the error.



- I  run the these codes before `bundle exec rspec`

```
rails db:create
rails db:schema:load
rails db:seed```

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️ 


@JuanLPalacios  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___

### ♻️ Request changes ♻️

- [x] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: spec/features/log_in_spec.rb (hunk 1 above)
Comment (JuanLPalacios) [reply spec/features/log_in_spec.rb]: can you help me, with 2 things:
first, can you provide me with the error log of the tests
second, can you run  in terminal:
```
rails c test
User.find(1)
```
and share the log

Comment (ahmetbozaci) [reply spec/features/log_in_spec.rb]: @JuanLPalacios Answered in the slack

**Comment by JuanLPalacios** [comment]
would kindly ask reviewers to to check if you seeded before running tests
```
rails db:create
rails db:schema:load
rails db:seed RAILS_ENV=test
```
and check that the test db has being seeded by 
```
rails c test
User.find(1)
```
if the return is nill the test db has not being seeded

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @JuanLPalacios👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**